pkg/integrations/node_exporter/config.go

```diff
@@ -300,209 +302,230 @@ func MapConfigToNodeExporterFlags(c *Config) (accepted []string, ignored []strin
+		DeviceExclude:   &c.EthtoolDeviceExclude,
+		IncludedMetrics: &c.EthtoolMetricsInclude,
 	}
 
-	if collectors[CollectorFilesystem] {
-		flags.add(
-			"--collector.filesystem.mount-timeout", c.FilesystemMountTimeout.String(),
-			"--collector.filesystem.mount-points-exclude", c.FilesystemMountPointsExclude,
-			"--collector.filesystem.fs-types-exclude", c.FilesystemFSTypesExclude,
-		)
+	cfg.Filesystem = collector.FilesystemConfig{
+		MountPointsExclude:     &c.FilesystemMountPointsExclude,
+		MountPointsExcludeSet:  true,
+		MountTimeout:           &c.FilesystemMountTimeout,
+		FSTypesExclude:         &c.FilesystemFSTypesExclude,
+		FSTypesExcludeSet:      true,
+		OldFSTypesExcluded:     &blankString,
+		OldMountPointsExcluded: &blankString,
+		StatWorkerCount:        &blankInt,
 	}
 
-	if collectors[CollectorIPVS] {
-		flags.add("--collector.ipvs.backend-labels", strings.Join(c.IPVSBackendLabels, ","))
+	var joinedLabels string
+	if len(c.IPVSBackendLabels) > 0 {
+		joinedLabels = strings.Join(c.IPVSBackendLabels, ",")
+		cfg.IPVS = collector.IPVSConfig{
+			Labels:    &joinedLabels,
+			LabelsSet: true,
+		}
+	} else {
+		cfg.IPVS = collector.IPVSConfig{
+			Labels:    &joinedLabels,
+			LabelsSet: false,
+		}
 	}
 
-	if collectors[CollectorNetclass] {
-		flags.addBools(map[*bool]string{
-			&c.NetclassIgnoreInvalidSpeedDevice: "collector.netclass.ignore-invalid-speed",
-		})
-
-		flags.add("--collector.netclass.ignored-devices", c.NetclassIgnoredDevices)
+	cfg.NetClass = collector.NetClassConfig{
+		IgnoredDevices: &c.NetclassIgnoredDevices,
+		InvalidSpeed:   &c.NetclassIgnoreInvalidSpeedDevice,
+		Netlink:        &blankBool,
+		RTNLWithStats:  &blankBool,
 	}
 
-	if collectors[CollectorNetdev] {
-		flags.addBools(map[*bool]string{
-			&c.NetdevAddressInfo: "collector.netdev.address-info",
-		})
-
-		flags.add(
-			"--collector.netdev.device-include", c.NetdevDeviceInclude,
-			"--collector.netdev.device-exclude", c.NetdevDeviceExclude,
-		)
+	cfg.NetDev = collector.NetDevConfig{
+		DeviceInclude:    &c.NetdevDeviceInclude,
+		DeviceExclude:    &c.NetdevDeviceExclude,
+		AddressInfo:      &c.NetdevAddressInfo,
+		OldDeviceInclude: &blankString,
+		OldDeviceExclude: &blankString,
+		Netlink:          &blankBool,
+		DetailedMetrics:  &blankBool,
 	}
 
-	if collectors[CollectorNetstat] {
-		flags.add("--collector.netstat.fields", c.NetstatFields)
+	cfg.NetStat = collector.NetStatConfig{
+		Fields: &c.NetstatFields,
 	}
 
-	if collectors[CollectorNTP] {
-		flags.add(
-			"--collector.ntp.server", c.NTPServer,
-			"--collector.ntp.protocol-version", fmt.Sprintf("%d", c.NTPProtocolVersion),
-			"--collector.ntp.ip-ttl", fmt.Sprintf("%d", c.NTPIPTTL),
-			"--collector.ntp.max-distance", c.NTPMaxDistance.String(),
-			"--collector.ntp.local-offset-tolerance", c.NTPLocalOffsetTolerance.String(),
-		)
-
-		flags.addBools(map[*bool]string{
-			&c.NTPServerIsLocal: "collector.ntp.server-is-local",
-		})
+	defaultPort := 123
```

Comment:
I feel that all these default should be imported directly from node exporter itself. Maybe we should add a TODO or something?